zebra/main.c

```diff
@@ -93,6 +93,7 @@ struct option longopts[] = {
 	{"retain", no_argument, NULL, 'r'},
 	{"vrfdefaultname", required_argument, NULL, 'o'},
 	{"graceful_restart", required_argument, NULL, 'K'},
+	{"skip_kernel", no_argument, NULL, 'X'},
```

Comment:
So let's start here: since you mentioned having trouble finding a single letter to use on the command-line, don't do that. Follow the example of the `--v6-rr-semantics` long option - use `--disable-kernel-install` or something like that, and you'll never have trouble with a single letter again.